test/02_Vault.ts

```diff
@@ -0,0 +1,129 @@
+import ERC20MockArtifact from '../artifacts/contracts/mocks/ERC20Mock.sol/ERC20Mock.json'
+import RateOracleMockArtifact from '../artifacts/contracts/mocks/RateOracleMock.sol/RateOracleMock.json'
+import SpotOracleMockArtifact from '../artifacts/contracts/mocks/SpotOracleMock.sol/SpotOracleMock.json'
+
+import { FYDai } from '../typechain/FYDai'
+import { Vault } from '../typechain/Vault'
+import { ERC20Mock } from '../typechain/ERC20Mock'
+import { RateOracleMock } from '../typechain/RateOracleMock'
+import { SpotOracleMock } from '../typechain/SpotOracleMock'
+
+import { ethers, waffle } from 'hardhat'
+import { expect } from 'chai'
+import { BigNumber } from 'ethers'
+const { deployContract, loadFixture } = waffle
+
+describe('Vault', function () {
+  let ownerAcc: SignerWithAddress
+  let userAcc: SignerWithAddress
+  let owner: string
+  let user: string
+  let fyDai: FYDai
+  let vault: Vault
+  let dai: ERC20Mock
+  let weth: ERC20Mock
+  let rateOracle: RateOracleMock
+  let spotOracle: SpotOracleMock
+  let maturity: BigNumber
+
+  const THREE_MONTHS = 3 * 30 * 24 * 60 * 60
+  const WAD = BigNumber.from(10).pow(18)
+  const ETH_SPOT = BigNumber.from(2000) // 1 ETH ~ 2000 DAI
+  const DAI_SPOT = WAD.div(BigNumber.from(2000)) // 1 DAI ~ 0.0005 ETH
+
+  async function fixture() {
+    const now = (await ethers.provider.getBlock(await ethers.provider.getBlockNumber())).timestamp
+    maturity = BigNumber.from(now + THREE_MONTHS)
+
+    rateOracle = (await deployContract(ownerAcc, RateOracleMockArtifact, [maturity])) as RateOracleMock
+    spotOracle = (await deployContract(ownerAcc, SpotOracleMockArtifact, [])) as SpotOracleMock
+
+    await spotOracle.setSpot(ETH_SPOT)
+
+    dai = (await deployContract(ownerAcc, ERC20MockArtifact, ['Dai', 'DAI'])) as ERC20Mock
+    weth = (await deployContract(ownerAcc, ERC20MockArtifact, ['Weth', 'WETH'])) as ERC20Mock
+    fyDai = (await deployContract(ownerAcc, FYDaiArtifact, [dai.address, maturity, 'Dai', 'DAI'])) as FYDai
+
+    await dai.mint(owner, WAD.mul(100))
+    await weth.mint(user, WAD.mul(10))
+    await fyDai.mint(owner, WAD.mul(100))
+
+    await dai.mint(fyDai.address, WAD.mul(100))
+
+    vault = (await deployContract(ownerAcc, VaultArtifact, [
+      dai.address,
+      weth.address,
+      spotOracle.address,
+      rateOracle.address,
+      [fyDai.address],
+    ])) as Vault
+  }
+
+  before(async () => {
+    const signers = await ethers.getSigners()
+    ownerAcc = signers[0]
+    userAcc = signers[1]
+    owner = await ownerAcc.getAddress()
+    user = await userAcc.getAddress()
+  })
+
+  beforeEach(async () => {
+    // loadFixture is a waffle feature that we use to load up the environment
+    // After the first execution, the blockchain is snapshotted, speeding up tests
+    // Incidentally, rolling back to the snapshot also rolls back in time
+    await loadFixture(fixture)
+  })
+
+  describe('without batch', async () => {
```

Comment:
This `describe` is not needed, and is not really true.

Reply:
The last commit should fix this.